src/kzg10/fastpoly.rs

```diff
@@ -0,0 +1,372 @@
+use crate::Error;
+use ark_ec::PairingEngine;
+use ark_ff::{Field, One, Zero};
+use ark_poly::UVPolynomial;
+
+use ark_poly::polynomial::univariate::DensePolynomial;
+
+type P<E: PairingEngine> = DensePolynomial<E::Fr>;
+
+/// Computes the inverse of f mod x^l
+pub fn inverse_mod_xl<E: PairingEngine>(f: &P<E>, l: usize) -> Option<P<E>> {
+    //use std::ops::Mul;
+    //let r =
+    //    std::mem::size_of::<u64>() * 8 - (l as u64).leading_zeros() as usize; // ceil(log_2(l))
+
+    //assert_eq!((l as f64).log2().ceil() as usize, r);
+    let r = (l as f64).log2().ceil() as usize; //TODO: rounding problems??
```

Comment:
I think `ark_std::log2()` should suffice for this.

Reply:
OK, I figured this had a good existing solution.